// vite-plugin-node/dist/vite-plugin-node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VitePluginNode = void 0;
const _1 = require(".");
const rollup_plugin_swc_1 = require("./rollup-plugin-swc");
const server_1 = require("./server");
function VitePluginNode(cfg) {
    var _a, _b;
    const config = {
        appPath: cfg.appPath,
        adapter: cfg.adapter,
        tsCompiler: (_a = cfg.tsCompiler) !== null && _a !== void 0 ? _a : 'esbuild',
        exportName: (_b = cfg.exportName) !== null && _b !== void 0 ? _b : 'viteNodeApp'
    };
    const plugins = [
        {
            name: _1.PLUGIN_NAME,
            config: () => ({
                server: {
                    hmr: false
                },
                esbuild: config.tsCompiler === 'esbuild' ? {} : false,
                VitePluginNodeConfig: config,
            }),
            configureServer: (server) => {
                server.middlewares.use(server_1.createMiddleware(server));
            },
            apply: 'serve'
        }
    ];
    if (config.tsCompiler === 'swc') {
        plugins.push({
            ...rollup_plugin_swc_1.RollupPluginSwc({
                jsc: {
                    target: 'es2019',
                    parser: {
                        syntax: 'typescript',
                        decorators: true,
                    },
                    transform: {
                        legacyDecorator: true,
                        decoratorMetadata: true,
                    },
                },
            })
        });
    }
    return plugins;
}
exports.VitePluginNode = VitePluginNode;
